model_analyzer/triton/server/server_docker.py

```diff
@@ -69,16 +71,39 @@ def start(self):
                 docker.types.DeviceRequest(device_ids=self._gpus,
                                            capabilities=[['gpu']])
             ]
-        environment = {
-            'CUDA_VISIBLE_DEVICES': ','.join([uuid for uuid in self._gpus])
-        }
 
-        # Mount required directories
-        volumes = {
-            self._server_config['model-repository']: {
-                'bind': self._server_config['model-repository'],
-                'mode': 'ro'
+        # Set environment inside container.
+        # Supports only strings, and value lookups/concats
+        env_cmds = []
+        if env:
+            env.update({
+                'CUDA_VISIBLE_DEVICES':
+                ','.join([uuid for uuid in self._gpus])
+            })
+
+            # Filter out variables that need to be set within container
+            # If the variable is LD_PRELOAD, it cannot be set in docker env
+            # If the variable contains a lookup, it must be done inside the container
+            for env_variable in list(env):
+                if env_variable == "LD_PRELOAD" or env[env_variable].find(
```

Comment:
What happens to the variables that do not satisfy this if condition?

Reply:
They are able to be passed into docker run's "environment" parameter.